### PR TITLE
SIDM-4188 The emailType header.

### PR DESCRIPTION
**Pull request by dfourn**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SIDM-4188

### Change description ###

Add the emailType header option for the stale user reregistration.

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/idam/api/fr/idm/selfservice/SelfServiceRegistrationApi.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
   @Headers({
     "Content-Type: application/json",
     "Accept: application/json",
+    "X-emailType: {emailType}"
```
Comment (radoslaw-orlowski-amido) [inline src/main/java/uk/gov/hmcts/reform/idam/api/fr/idm/selfservice/SelfServiceRegistrationApi.java:37]: Have these changes been generated using `gradle generateApi` or manually?

Comment (dfourn) [reply src/main/java/uk/gov/hmcts/reform/idam/api/fr/idm/selfservice/SelfServiceRegistrationApi.java:37]: manual

Comment (radoslaw-orlowski-amido) [reply src/main/java/uk/gov/hmcts/reform/idam/api/fr/idm/selfservice/SelfServiceRegistrationApi.java:37]: Can you please regenerate the api automatically from the yaml files (`gradle generateApi`) then copy the relevant part to the actual code from `build/src/generated`? Thanks

Comment (tbamido) [reply src/main/java/uk/gov/hmcts/reform/idam/api/fr/idm/selfservice/SelfServiceRegistrationApi.java:37]: yeah I think some things will change if we generate automatically, order of fields maybe, not sure if it's a big deal

Comment (radoslaw-orlowski-amido) [reply src/main/java/uk/gov/hmcts/reform/idam/api/fr/idm/selfservice/SelfServiceRegistrationApi.java:37]: If you are talking about existing code, then that would only mean that there are manual changes there already which should not be there in the first place. But yes, the regeneration can potentially change the order of the params. Easy to spot though as tests will start failing :)

Comment (dfourn) [reply src/main/java/uk/gov/hmcts/reform/idam/api/fr/idm/selfservice/SelfServiceRegistrationApi.java:37]: regenerated but also modified the generated code

**Pull request merged**